Explorer/Assets/DCL/Infrastructure/SceneRuntime/Apis/Modules/SignedFetch/SignedFetchWrap.cs

```diff
@@ -25,28 +27,80 @@ public class SignedFetchWrap : IJsApiWrapper
             IDecentralandUrlsSource decentralandUrlsSource,
             ISceneData sceneData,
-            IRealmData realmData)
+            IRealmData realmData,
+            IWeb3IdentityCache identityCache)
         {
             this.webController = webController;
             this.decentralandUrlsSource = decentralandUrlsSource;
             this.sceneData = sceneData;
             this.realmData = realmData;
+            this.identityCache = identityCache;
+        }
+
+        public void Dispose()
+        {
+            cancellationTokenSource.SafeCancelAndDispose();
         }
 
         [UsedImplicitly]
-        public object Headers(SignedFetchRequest signedFetchRequest)
+        public object GetSignedHeaders(string url, string body, string headers, string method)
         {
-            string jsonMetaData = signedFetchRequest.init?.body ?? string.Empty;
+            Dictionary<string, string>? deserializedHeaders = JsonConvert.DeserializeObject<Dictionary<string, string>>(headers);
+
+            string response = GetSignedHeaders(new SignedFetchRequest
+            {
+                url = url,
+                init = new FlatFetchInit
+                {
+                    body = body,
+                    headers = deserializedHeaders ?? new Dictionary<string, string>(),
+                    method = string.IsNullOrEmpty(method) ? "get" : method,
+                },
+            });
+
+            return response;
+        }
+
+        private string GetSignedHeaders(SignedFetchRequest request)
+        {
+            string? method = request.init?.method?.ToLower();
+            ulong unixTimestamp = DateTime.UtcNow.UnixTimeAsMilliseconds();
+            string? hashBody = null;
+
+            if (request.init != null)
+                if (!string.IsNullOrEmpty(request.init.body))
+                    hashBody = sha256_hash(request.init.body);
 
-            return new WebRequestHeadersInfo()
-                  .WithSign(jsonMetaData, DateTime.UtcNow.UnixTimeAsMilliseconds())
-                  .AsMutableDictionary();
+            string signatureMetadata = CreateSignatureMetadata(hashBody);
+
+            var headers = new WebRequestHeadersInfo(request.init?.headers)
+                         .WithSign(signatureMetadata, unixTimestamp)
+                         .AsMutableDictionary();
+
+            var signInfo = WebRequestSignInfo.NewFromRaw(
+                signatureMetadata,
+                request.url,
+                unixTimestamp,
+                method ?? string.Empty
+            );
+
+            using AuthChain authChain = identityCache.EnsuredIdentity().Sign(signInfo.StringToSign);
+            var authChainIndex = 0;
+
+            foreach (AuthLink link in authChain)
+            {
+                headers[$"x-identity-auth-chain-{authChainIndex}"] = link.ToJson();
+                authChainIndex++;
+            }
+
+            return JsonConvert.SerializeObject(headers);
```

Comment:
It would be much more efficient to build the json string directly instead of allocating objects and then serializing them to json.

Reply:
Actually, can we write directly to the network stream instead of allocating a string?

Reply:
I want you to always think: what is the final destination of this data, and can I just skip everything and write it directly there.

Reply:
In this case, the absolute best would be to allocate an StdString, obtain a pointer to its internal buffer, wrap that in an UnmanagedMemoryStream and use JsonWriter to write directly to it.

Reply:
A lot of work, admittedly, which is why my first question was "how often is this called".